### DISCUSSION TITLE
Impossible to install a model with Poetry

### DISCUSSION THREAD

**Discussion by LifeIsStrange** [explosion/spaCy#10278] [Help: Installation] [answered]
poetry add spacy works
it resolves to 
setuptools = "^60.8.2"
wheel = "^0.37.1"
spacy = "^3.2.2" in my pyproject.toml however
en_core_web_trf = { url = "https://github.com/explosion/spacy-models/releases/download/en_core_web_trf-3.2.0/en_core_web_trf-3.2.0-py3-none-any.whl" }
or poetry add https://github.com/explosion/spacy-models/releases/download/en_core_web_trf-3.2.0/en_core_web_trf-3.2.0-py3-none-any.whl
leads to the following error:
```Unable to find installation candidates for torch (1.10.2)```

I don't know what to do, it worked on older versions. Spacy **should** work with Poetry, I have lost a day already with this.
On a fresh virtual env, with python 3.10
@adrianeboyd friendly ping help plz

**Comment by LifeIsStrange** [comment]
https://stackoverflow.com/questions/56239310/could-not-find-a-version-that-satisfies-the-requirement-torch-1-0-0
so spacy would silently include pytorch 1.10.2 which does not support python past 3.7.9? ...

**Comment by polm** [comment, marked as answer]
Please do not ping individual maintainers, it disrupts notifications. 

I'm not exactly sure why this isn't working, but it may help to know that the torch dependency isn't silent - it comes from spacy-transformers. Also the answer you link to is from October 2020, and while Pytorch is a little slow to support new Python versions, at least 3.9 is supported without issue. (3.10 seems to have a Linux wheel out but may be a work in progress, see https://github.com/pytorch/pytorch/issues/67669).

It might help if you include your full Poetry file - for example, you haven't clarified what version of Python you're trying to use.

**Reply by LifeIsStrange** [reply]
Yes the issue was 3.10 related, thanks
